core/src/banking_stage.rs

```diff
@@ -589,12 +589,13 @@ impl BankingStage {
         macro_rules! spawn_scheduler {
             ($scheduler:ident) => {
                 let exit = exit.clone();
+                let shutdown_signal = self.banking_shutdown_signal.clone();
                 let bank_forks = self.bank_forks.clone();
                 threads.push(
                     Builder::new()
                         .name("solBnkTxSched".to_string())
                         .spawn(move || {
-                            let scheduler_controller = SchedulerController::new(
+                            let mut scheduler_controller = SchedulerController::new(
```

Comment:
Q: is this change necessary for signaling manager to exit?

Reply:
Yes, we don't want manager to be dropped which will cause the workers to be signalled that the channel has been closed. We must ensure our manager thread observes the cancellation before it observes the workers exiting due to the channel closure

Reply:
Ah, I see - `scheduler_controller` must be dropped after `shutdown_signal.cancel()`, subtle.